Proton/Sources/ObjC/PRTextStorage.m

```diff
@@ -97,17 +97,17 @@ - (void)replaceCharactersInRange:(NSRange)range withAttributedString:(NSAttribut
     }
 
     NSAttributedString *deletedText = [_storage attributedSubstringFromRange:range];
-    [_textStorageDelegate textStorage:self willDeleteText:deletedText insertedText:replacementString range:range];
+    [_textStorageDelegate textStorage:self will:deletedText insertText:replacementString in:range];
     [super replaceCharactersInRange:range withAttributedString:replacementString];
 }
 
 - (void)replaceCharactersInRange:(NSRange)range withString:(NSString *)str {
     [self beginEditing];
     NSInteger delta = str.length - range.length;
 
-    NSArray<Attachment *> *attachmentsToDelete = [self attachmentsForRange:range];
-    for (Attachment *attachment in attachmentsToDelete) {
-        [attachment removeFromSuperview];
+    NSArray<NSTextAttachment *> *attachmentsToDelete = [self attachmentsForRange:range];
+    for (NSTextAttachment *attachment in attachmentsToDelete) {
+        [_textStorageDelegate textStorage:self didDelete:attachment];
```

Comment:
removal of attachment view is now delegated to `RichTextView` as `PRTextStorage` does not have knowledge of `Attachment` anymore.